### PR TITLE
s4909464 Graph Neural Network 

### PR DESCRIPTION
**Pull request by wuzhe1234567**
Author: Zhe Wu （student ID: 49094642)
Task: Normal Difficulty

This pull request shows a semi-supervised multi-class node classification on Facebook large page network dataset by using CNN model. The project outputs visualization images and prints the accuracy to see the performance of the model. 

The directory has the following structure：
- ```README.md ```: contains a detailed description of the model
- ```dataset.py``` : contains loading and processing of datasets
- ```modules.py``` : contains GCN model
- ```train.py``` : contains model training and saving
- ```predict.py``` : contains the model's predictions for the data
- ```utils.py``` : contains visualization results after data processing
- images/ : directory, contains the graphs
- outputs/ : directory, cotains the results

### CODE REVIEW

**Comment by gayanku** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Normal

Readme Overall: Good

- Project Overview: Good
- Model Documented/Explained: Good
- Data + Pre-processing: Not done. Used 128 embedding as model input. This is ok.
- Training / Loss Curve(s): Good.
- Result Demonstration: Good
- References: Good
- Setup/Reproducibility/Requirements: Good

Functionality:

- Multi-layer GCN: Good, 4 layers. Uses external library (torch_geometric.nn GCNConv). 
- Semi supervised: Good
- Accuracy: Good.
- TSNE or UMAP: Good, 
- Discussion: Good

Code: 
- Consistent with Results: OK.
- File Structure: Good
- Commenting: Good
- Commit frequency: Good
- Commit messages: Good
- Reusable Modular code/Design: Good

Other Comments:
Good best acc based model selection.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. Some overfitting seen. No early stop.</td><td>-1</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Normal. GNN Task.</td><td>-5</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Good. </td><td></td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). </td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is adequate. Give a  high level intro to the approach instead of the file list.</td><td>-1</td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-8</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by shakes76** [comment]
Meaningful commit messages -1
